pkg/provisioner/ironic/ironic.go

```diff
@@ -2194,3 +2194,42 @@ func (p *ironicProvisioner) RemoveBMCEventSubscriptionForNode(subscription metal
 	}
 	return operationComplete()
 }
+
+// Fetches the VirtualMedia details of the BareMetalHost and updates DataImage.
+func (p *ironicProvisioner) GetDataImageStatus() (nodeReservation string, nodeLastError string) {
+	// TODO(hroyrh)
+	// Get BareMetalHost VirtualMedia details and handle errors
+
+	node, _ := p.getNode()
```

Comment:
Please don't ignore errors like this. Especially don't just proceed with the rest of the logic: you'll get empty Reservation and LastError on error.

Reply:
Fixed